lib/compat/wordpress-6.8/site-editor.php

```diff
@@ -116,9 +116,33 @@ function gutenberg_redirect_site_editor_deprecated_urls() {
  * @return callable The default handler or a custom handler.
  */
 function gutenberg_styles_wp_die_handler( $default_handler ) {
-	if ( ! wp_is_block_theme() && str_contains( $_SERVER['REQUEST_URI'], 'site-editor.php' ) && isset( $_GET['p'] ) ) {
+	if ( ! wp_is_block_theme() && str_contains( $_SERVER['REQUEST_URI'], 'site-editor.php' ) && current_user_can( 'edit_theme_options' ) ) {
 		return '__return_false';
 	}
 	return $default_handler;
 }
 add_filter( 'wp_die_handler', 'gutenberg_styles_wp_die_handler' );
+
+/**
+ * Add a Styles submenu under the Appearance menu
+ * for Classic themes.
+ *
+ * @global array $submenu
+ */
+function gutenberg_add_styles_submenu_item() {
+	if ( ! wp_is_block_theme() && current_theme_supports( 'editor-styles' ) ) {
```

Comment:
I don't think there's any need to check `editor-styles`  support. There are probably themes that configure editor styles only in `theme.json`.

For example, if you enable the Emptyhybrid theme in the `http://localhost:8889/` environment, the Design submenu will not be displayed.

Reply:
Oh that's a good point, if a hybrid theme were using _only_ theme.json for its styles, it wouldn't need an editor stylesheet. I think in that case we need to check for the presence of one or the other, because if the theme has neither, the style book will be useless.